src/steps/fetch.py

```diff
@@ -58,3 +59,24 @@ def update_repo(clone_dir):
     command = 'git pull --unshallow'
 
     return run(logger, command, cwd=clone_dir)
+
+
+def fetch_commit_sha(clone_dir):
+    '''
+    fetch the last commitSHA
+    '''
+    logger = get_logger('clone')
+    logger.info('Fetching commit details')
+    command = shlex.split('git log -1')  # get last commit only
+    process = subprocess.run(  # nosec
+        command,
+        shell=False,
+        check=True,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+        cwd=clone_dir
+    )
+    commit_log = process.stdout
+    logger.info(commit_log)  # display last commit details in log
+    commit_sha = shlex.split(commit_log)[1]
```

Comment:
[`shlex.split`](https://docs.python.org/3/library/shlex.html) is specifically for splitting a string into syntax understandable for shells. If you want to just split on a space, let's do it the conventional way: `commit_log.split()`.